sphinx/ext/napoleon/__init__.py

```
@@ -378,6 +382,140 @@ def _process_docstring(app: Sphinx, what: str, name: str, obj: Any,
     lines[:] = result_lines[:]
 
 
+def _unwrap_callable(obj: Any) -> Any:
+    try:
+        return inspect.unwrap(obj)  # type: ignore[arg-type]
+    except Exception:
```

[major] Catching every Exception here silently papers over real bugs in the unwrap chain. Please narrow this to the specific failure modes we expect from inspect.unwrap (e.g. ValueError for wrap cycles) so unexpected exceptions surface instead of being swallowed.